Bundle/WidgetMapBundle/Manager/WidgetMapManager.php

```diff
@@ -47,185 +101,153 @@ public function updateWidgetMapOrder(View $view, $sortedWidget)
-        } elseif ($parentWidgetMapEntry = $slot->getWidgetMapByWidgetId($parentWidgetId)) {
-            // Place the widget just under the parent widget
-            $widgetMapEntry = new WidgetMap();
-            $widgetMapEntry->setPosition($parentWidgetMapEntry->getPosition() + 1);
-            $widgetMapEntry->setAction($originalWidgetMap->getAction());
-            $widgetMapEntry->setWidgetId($widgetId);
-            $widgetMapEntry->setPositionReference($parentWidgetMapEntry->getPositionReference());
-
-        // If the parent of the sorted widget is not from the current page
-        } else {
-            $widgetMapEntry = new WidgetMap();
-            $widgetMapEntry->setPosition(1);
-            $widgetMapEntry->setAction($originalWidgetMap->getAction());
-            $widgetMapEntry->setPositionReference($parentWidgetId);
-            $widgetMapEntry->setWidgetId($widgetId);
-        }
-
-        // If this WidgetMapEntry already in the page, remove it
-        if ($oldWidgetMapEntry = $slot->getWidgetMapByWidgetId($widgetId)) {
-            $widgetMapEntry->setAsynchronous($oldWidgetMapEntry->isAsynchronous());
-            $slot->removeWidgetMap($oldWidgetMapEntry);
-        // Else, the new widgetMap is an overwrite
-        } elseif ($originalWidgetMap->getAction() !== WidgetMap::ACTION_OVERWRITE) {
-            $widgetMapEntry->setAction(WidgetMap::ACTION_OVERWRITE);
-            $widgetMapEntry->setReplacedWidgetId($widgetId);
-        }
-        // Insert the new one in page slot
-        $this->helper->insertWidgetMapInSlot($slotId, $widgetMapEntry, $view);
+        $widgetMap = new WidgetMap();
+        $widgetMap->setAction(WidgetMap::ACTION_OVERWRITE);
+        $widgetMap->setReplaced($originalWidgetMap);
+        $widgetMap->setWidget($widgetCopy);
+        $widgetMap->setView($view);
+        $widgetMap->setSlot($originalWidgetMap->getSlot());
+        $widgetMap->setPosition($originalWidgetMap->getPosition());
+        $widgetMap->setAsynchronous($widgetCopy->isAsynchronous());
+        $widgetMap->setParent($originalWidgetMap->getParent());
 
-        return;
+        $view->addWidgetMap($widgetMap);
     }
 
     /**
-     * create a widgetMap for the new Widget cloned.
+     * If the moved widgetMap has someone at both his before and after, arbitrary move UP the before side
+     * and find the first place after the before widgetMap hierarchy to place the after widgetMap.
      *
-     * @return void
-     **/
-    public function overwriteWidgetMap(Widget $widgetCopy, Widget $widget, View $view)
+     * @param View $view
+     * @param $beforeChild
+     * @param $afterChild
+     * @param $originalParent
+     * @param $originalPosition
+     */
+    public function moveChildren(View $view, $beforeChild, $afterChild, $originalParent, $originalPosition)
     {
-        //the id of the new widget
-        $widgetId = $widgetCopy->getId();
+        if ($beforeChild && $afterChild) {
```

Comment:
seems to be same code part of move function https://github.com/paulandrieux/victoire/blob/feature/refactor-dnd/Bundle/WidgetMapBundle/Manager/WidgetMapManager.php#L69